src/bin/lib/subcommands.c/runprogram.h

```diff
@@ -77,51 +80,55 @@ static void waitprogram(Program *prog, pid_t childPid);
  * the run and then return a Program struct instance with the result of running
  * the program.
  */
-Program
+Program *
```

Comment:
I did not fully understand why we wanted to change the return type of this function.

Do we want the return value to be automatically garbage collected for some reason?

Reply:
The static structure instance allocated within the `run_program` function is not safe to use from the outside. It's an overdue fix that fitted well with this PR changes.